File: hw6/hw6.py
class Shape:
    def __init__(self, color):
        self.__color = color
    def get_color(self):
        return self.__color

class Circle(Shape):
    def __init__(self, color, radius):
        super().__init__(color)
        self.__radius = radius
    def get_radius(self):
        return self.__radius
    def get_area(self):
        r = self.get_radius()
        area = 3.14159265359 * r* r
        return area
    def print(self):
        c_r = self.get_radius()
        c_c = self.get_color()
        c_a = self.get_area()
        print(c_c, ' circle, radius ', c_r,', area ', c_a)

class Rectangle(Shape):
    def __init__(self, color, heigth, width):
        super().__init__(color)
        self.__height = heigth
        self.__width = width
    def get_height(self):
        return self.__height
    def get_width(self):
        return self.__width
    def get_area(self):
        h = self.get_height()
        w = self.get_width()
        area = h * w
        return area
    def print(self):
        r_c = self.get_color()
        r_h = self.get_height()
        r_w = self.get_width()
        r_a = self.get_area()
        print(r_c, ' rectangle, height ', r_h,', width ', r_w, ', area ', r_a)

class Triangle(Shape):
    def __init__(self, color, height, base):
        super().__init__(color)
        self.__height = height
        self.__base = base
    def get_height(self):
        return self.__height
    def get_base(self):
        return self.__base
    def get_area(self):
        h = self.get_height()
        b = self.get_base()
        area = h * b / 2
        return area
    def print(self):
        t_c = self.get_color()
        t_h = self.get_height()
        t_b = self.get_base()
        t_a = self.get_area()
        print(t_c, ' triangle, height ', t_h,', base ', t_b, ', area ', t_a)

test1 = Circle('green', 10)
test1.print()

test2 = Rectangle('red', 3, 4)
test2.print()

test3 = Triangle('pink', 5, 7)
test3.print()

File: hw6/hw6.java
import java.io.*;
import java.util.*;

abstract class Shape{
    protected String color;
    public Shape(String inp){
        color = inp;
    }
    public abstract double get_area();
    public void print(){
        //
    }
    public String get_color(){
        return color;
    }
}

class Circle extends Shape{
    private int radius;
    public Circle(String color, int r){
        super(color);
        radius = r;
    }
    public void print(){
       double area = 0;
       String color;
       area = get_area();
       color = get_color();
       System.out.println(color + " circle, radius " + radius + ", area " + area);
    }
    public double get_area(){
        double area = 0;
        area = 3.141592653559 * radius * radius;
        return area;
    }
}

class Rectangle extends Shape{
    private int height;
    private int width;
    public Rectangle(String color, int h, int w){
        super(color);
        height = h;
        width = w;
    }
    public void print(){
        double area = 0;
        String color;
        area = get_area();
        color = get_color();
        System.out.println(color + " rectangle, height " + height + ", width " + width + ", area " + area);
    }
    public double get_area(){
        double area = 0;
        area = height * width;
        return area;
    }
}

class Triangle extends Shape{
    private int height;
    private int base;
    public Triangle(String color, int h, int b){
        super(color);
        height = h;
        base = b;
    }
    public void print(){
        double area = 0;
        String color;
        area = get_area();
        color = get_color();
        System.out.println(color + " triangle, height " + height + ", base " + base + ", area " + area);
    }
    public double get_area(){
        double area = 0;
        area = height * base / 2;
        return area;
    }
}

public class hw6{
    public static void main(String[] args)throws IOException{
        //Shape s;
        //s = new Triangle("green", 3, 4);
        //s.print();

        //Circle b = new Circle("green", 5);
        //b.print();

        //Rectangle c = new Rectangle("pink", 2, 4);
        //c.print();

        Vector<Shape> shape = new Vector<Shape>();
        Vector<String> type = new Vector<String>();
        String ss = "yellow";
        int x = 10, y = 20;
        Shape circle = new Circle(ss, x);
        Shape a = new Triangle(ss, x, y);
        Shape b = new Rectangle(ss, x, y);
        Shape c = new Triangle("red", 3, 5);

        Shape d = new Circle("green", 5);
        Shape e = new Rectangle("white", 3, 8);

        shape.addElement(circle);
        type.addElement("circle");
        shape.addElement(a);
        type.addElement("triangle");
        shape.addElement(b);
        type.addElement("rectangle");
        shape.addElement(c);
        type.addElement("triangle");
        shape.addElement(d);
        type.addElement("circle");
        shape.addElement(e);
        type.addElement("rectangle");

        System.out.println("Printing all shapes:");
        for(Shape p:shape){
            p.print();
        }

    }
}